test/compress/issue-1770.js

```diff
@@ -62,21 +46,68 @@ mangle_props: {
             obj[-1/0],
             obj[-1/0],
             obj["-Infinity"],
+            obj[null],
+            obj["null"]
+        );
+    }
+    expect_stdout: "1 1 1 2 2 2 3 3 3 4 4 4 5 5"
+}
+
+numeric_literal: {
+    mangle_props = {}
+    input: {
+        var obj = {
+            "-0": 1,
+            42: 2,
+            "42": 3,
+            0x25: 4,
+            "0x25": 5,
+            1E42: 6,
+            "1E42": 7,
+            "1e+42": 8,
+        };
+        console.log(
             obj[-0],
             obj[-""],
             obj["-0"],
-            obj[null],
-            obj["null"],
+            obj[42],
+            obj["42"],
+            obj[0x25],
+            obj["0x25"],
+            obj[37],
+            obj["37"],
+            obj[1E42],
+            obj["1E42"],
+            obj["1e+42"]
+        );
+    }
+    expect: {
+        var obj = {
+            "-0": 1,
+            42: 2,
+            "42": 3,
+            37: 4,
+            a: 5,
+            1E42: 6,
+            b: 7,
+            "1e+42": 8,
+        };
+        console.log(
+            obj[-0],
+            obj[-""],
+            obj["-0"],
+            obj[42],
+            obj["42"],
             obj[37],
             obj["a"],
             obj[37],
             obj["37"],
-            obj[1e100],
+            obj[1E42],
```

Comment:
Done in https://github.com/mishoo/UglifyJS2/pull/1779/commits/5e76e8d0159697000c5a0a4404f9ae7e9ed29b25